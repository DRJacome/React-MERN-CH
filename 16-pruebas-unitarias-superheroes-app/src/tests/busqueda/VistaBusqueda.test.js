import React from 'react'
import { mount } from 'enzyme'
import { VistaBusqueda } from '../../components/busqueda/VistaBusqueda';
import { MemoryRouter, Route } from 'react-router-dom';

describe('Pruebas en <VistaBusqueda/>', () => {

  test('Debe mostrar el componente correctamente con valores por defecto', () => {
    const wrapper = mount(
      <MemoryRouter initianEntries={['/search']}>
        <Route path="/search" component={VistaBusqueda} />
      </MemoryRouter>
    );

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('.alert-info').text().trim()).toBe('Tu búsqueda aparecerá aquí');
  })

  test('Debe mostrar "batman" y el input con el valor del queryString', () => {
    const wrapper = mount(
      <MemoryRouter initianEntries={['/search?=batman']}>
        <Route path="/search" component={VistaBusqueda} />
      </MemoryRouter>
    );

    expect(wrapper.find('input').prop('value')).toBe('batman');
    expect(wrapper).toMatchSnapshot();
  })

  test('Debe mostrar un error si no se encuentra el héroe', () => {
    const wrapper = mount(
      <MemoryRouter initianEntries={['/search?=batman123']}>
        <Route path="/search" component={VistaBusqueda} />
      </MemoryRouter>
    );

    expect(wrapper.find('.alert-danger').text().trim()).toBe('No hay ningún superhéroe llamado batman123; prueba con otra búsqueda.');
    expect(wrapper).toMatchSnapshot();
  })

  test('Debe llamar el push del history', () => {
    const historial = {
      push: jest.fn()
    }

    const wrapper = mount(
      <MemoryRouter initianEntries={['/search?=batman123']}>
        <Route path="/search" component={() => <VistaBusqueda history={historial} />} />
      </MemoryRouter>
    );

    wrapper.find('input').simulate('change', {
      target: {
        name: 'textoBusqueda',
        value: 'batman'
      }
    });

    wrapper.find('form').prop('onSubmit')({
      preventDefault() { }
    });

    expect(historial.push).toHaveBeenCalledWith('?q=batman');
  })
})