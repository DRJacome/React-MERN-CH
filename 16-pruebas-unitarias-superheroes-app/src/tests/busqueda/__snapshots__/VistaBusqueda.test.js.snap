// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <VistaBusqueda/> Debe mostrar el componente correctamente con valores por defecto 1`] = `""`;
